x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/enrich/EnrichPolicyResolver.java

```diff
@@ -471,11 +477,7 @@ protected Map<String, EnrichPolicy> availablePolicies() {
         return projectResolver.getProjectMetadata(clusterService.state()).custom(EnrichMetadata.TYPE, EnrichMetadata.EMPTY).getPolicies();
     }
 
-    protected void getRemoteConnection(String cluster, ActionListener<Transport.Connection> listener) {
-        remoteClusterService.maybeEnsureConnectedAndGetConnection(
-            cluster,
-            remoteClusterService.isSkipUnavailable(cluster).orElse(true) == false,
-            listener
-        );
+    protected void getRemoteConnection(String cluster, EsqlExecutionInfo executionInfo, ActionListener<Transport.Connection> listener) {
+        remoteClusterService.maybeEnsureConnectedAndGetConnection(cluster, executionInfo.shouldSkipOnFailure(cluster) == false, listener);
```

Comment:
Just to ensure - the desired behavior on CPS is that we always want to try to reconnect immediately if the connection is bad, when `allow_partial = true`, and don't try to reconnect when `allow_partial = false`?